### PR TITLE
refactor: fix issues reported by staticcheck and enable linter

### PR DESCRIPTION
**Pull request by sbchaos**

### CODE REVIEW

File: api/handler/v1beta1/adapter.go [hunk 1]
```diff
@@ -8,11 +8,11 @@ import (
 
 	"google.golang.org/protobuf/types/known/durationpb"
 
-	"github.com/golang/protobuf/proto"
```
Comment (sbchaos) [inline api/handler/v1beta1/adapter.go:11]: This package is deprecated

File: api/handler/v1beta1/project.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func (sv *RuntimeServiceServer) RegisterProject(ctx context.Context, req *pb.Reg
 	}
 
 	responseMsg := "project saved successfully."
-	if req.Namespace != nil {
+	if req.Namespace != nil { //nolint:staticcheck
```
Comment (sbchaos) [inline api/handler/v1beta1/project.go:17]: Silence lint error for using deprecated field. Used only to show a message.

File: cmd/server/server.go [hunk 3]
```diff
@@ -563,7 +563,7 @@ func Initialize(l log.Logger, conf config.Optimus) error {
 	}
 
 	// We'll accept graceful shutdowns when quit via SIGINT (Ctrl+C)
-	signal.Notify(termChan, os.Interrupt, os.Kill, syscall.SIGTERM)
```
Comment (sbchaos) [inline cmd/server/server.go:566]: We cannot interrupt `os.Kill`. Already catching `syscall.SIGTERM`

File: extension/extension_test.go [hunk 4]
```diff
@@ -59,7 +59,7 @@ func (e *ExtensionSuite) TestInstall() {
 			panic(err)
 		}
 
-		err = ext.Install(nil, e.validOwner, e.validRepo, e.validAlias)
+		err = ext.Install(nil, e.validOwner, e.validRepo, e.validAlias) // nolint:staticcheck
```
Comment (sbchaos) [inline extension/extension_test.go:62]: Test for checking if context is nil

File: models/plugin.go [hunk 5]
```diff
@@ -139,7 +139,7 @@ type PluginQuestions []PluginQuestion
 
 func (q PluginQuestions) Get(name string) (PluginQuestion, bool) {
 	for _, que := range q {
-		if strings.ToLower(que.Name) == strings.ToLower(name) {
+		if strings.EqualFold(que.Name, name) {
```
Comment (sbchaos) [inline models/plugin.go:142]: Using case insensitive comparison, to make intention more clear

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**